octobot_commons/databases/cache_manager.py

```diff
@@ -0,0 +1,178 @@
+                self.__class__.CACHES.set_node_at_path(cache, None, cache_path)
+                return cache.get_database()
+            raise common_errors.NoCacheValue(f"Cache is initialized for {tentacle_name} on {exchange_name} "
+                                             f"{symbol} {time_frame}")
+
+    def has_cache(self, tentacle_name, exchange_name, symbol, time_frame, config_name=None):
+        identifier = config_name or self.DEFAULT_CONFIG_IDENTIFIER
+        try:
+            return bool(self.__class__.CACHES.get_node([tentacle_name, exchange_name, symbol, time_frame, identifier]))
+        except event_tree.NodeExistsError:
+            return False
+
+    def get_cache_registered_requirements(self, tentacle_name, exchange_name, symbol, time_frame, config_name=None):
+        identifier = config_name or self.DEFAULT_CONFIG_IDENTIFIER
+        return self.__class__.CACHES.get_node([tentacle_name, exchange_name, symbol, time_frame, identifier]).\
+            node_value.tentacles_requirements
+
+    def get_cache_previous_db_metadata(self, tentacle_name, exchange_name, symbol, time_frame, config_name=None):
+        identifier = config_name or self.DEFAULT_CONFIG_IDENTIFIER
+        return self.__class__.CACHES.get_node([tentacle_name, exchange_name, symbol, time_frame, identifier]).\
+            node_value.previous_db_metadata
+
+    async def clear_cache(self, tentacle_name, exchange_name=None, symbol=None, time_frame=None, config_name=None):
+        try:
+            for cache, _ in self._caches(tentacle_name, exchange_name, symbol, time_frame, config_name):
+                await cache.node_value.clear()
+            return True
+        except event_tree.NodeExistsError:
+            return False
+
+    async def reset_cache(self, tentacle_name, exchange_name, symbol, time_frame, config_name):
+        identifier = config_name or self.DEFAULT_CONFIG_IDENTIFIER
+        cache = self.__class__.CACHES.delete_node([tentacle_name, exchange_name, symbol, time_frame, identifier])
+        await cache.node_value.close()
+
+    async def close_cache(self, tentacle_name, exchange_name=None, symbol=None, time_frame=None, config_name=None,
+                          reset_cache_db_ids=False):
+        try:
+            to_remove_caches = []
+            for cache, identifiers in self._caches(tentacle_name, exchange_name, symbol, time_frame, config_name):
+                await cache.node_value.close()
+                to_remove_caches.append(identifiers)
+            if reset_cache_db_ids:
+                # remove cache from caches to force complete reopen of the cache db
+                # (might be at a different place)
+                for identifier in to_remove_caches:
+                    self.__class__.CACHES.delete_node(identifier)
+            return True
+        except event_tree.NodeExistsError:
+            return False
+
+    async def reset(self):
+        for cache, _ in self._caches(None, None, None, None, None):
```

Comment:
Why None shouldn't be the default value of caches arguments?

Reply:
It is now :)